ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/snapsync/SnapWorldDownloadState.java

```diff
@@ -271,7 +279,8 @@ public synchronized void reloadTrieHeal() {
   }
 
   public synchronized void startFlatDatabaseHeal(final BlockHeader header) {
-    LOG.info("Initiating the healing process for the flat database");
+    syncDurationMetrics.startTimer(SyncDurationMetrics.Labels.FLAT_DB_HEAL);
+    syncDurationMetrics.stopTimer(SyncDurationMetrics.Labels.CHAIN_DOWNLOAD_DURATION);
```

Comment:
this is a surprising place to find this imo.  If a node starts and does not perform a worldstate sync/heal, will the chain download duration count upwards endlessly?  IIRC the chain download pipeline doesn't ever stop until the node stops.

Perhaps a better place to stop this timer would be as an InitialSyncListener in PipelineChainDownloader.

Reply:
You are right, that is surprising. The problem with the PipelineChainDownloader is that it is started and stopped multiple times. I have made clear in the PR description that these metrics are only for snap and cp sync. Fast sync is broken at the moment and we will probably deprecate it once snap works for private networks.
I have moved the stopTimer out of the startFlatDatabaseHeal, as you can still configure besu to not do it.